src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -203,6 +203,31 @@ public function findByIds(array $ids): array
             ->getQuery()->getResult(Query::HYDRATE_ARRAY);
     }
 
+    public function findProgrammesByAncestryAndType(array $ancestryDbIds, string $entityType, ?int $limit, int $offset) : array
+    {
+        $this->assertEntityType($entityType, ['Clip', 'Episode', 'Series']);
+
+        $qb = $this->createQueryBuilder('programme')
+                   ->addSelect(['masterBrand', 'image', 'mbImage'])
+                   ->leftJoin('programme.image', 'image')
+                   ->leftJoin('programme.masterBrand', 'masterBrand')
+                   ->leftJoin('masterBrand.image', 'mbImage')
+                   ->where("programme INSTANCE OF ProgrammesPagesService:$entityType")
```

Comment:
I don't think you can use a parameter there. Everything else in this class also does this.